src/lib/rest/rest.cpp

```diff
@@ -614,6 +614,65 @@ bool urlCheck(ConnectionInfo* ciP, const std::string& url)
   return true;
 }
 
+/* ****************************************************************************
+*
+* defaultServicePath
+*
+* Returns a default servicePath (to be used in the case Fiware-servicePath header
+* is not found in the request) depending on the RequestCheck
+*/
+std::string defaultServicePath(const char* url, const char* method)
+{
+  /* Look for standard operation (based on the URL). Given that some operations are
+   * a substring of other, we search first for the longest ones
+   *
+   *  updateContextAvailabilitySubscription
+   *  unsubscribeContextAvailability
+   *  subscribeContextAvailability
+   *  discoverContextAvailability
+   *  updateContextSubscription
+   *  unsubscribeContext
+   *  subscribeContext
+   *  registerContext
+   *  updateContext
+   *  queryContext
+   *
+   * Note that in the case of unsubscribeContext and unsubscribeContextAvailability doesn't
+   * actually use a servicePath, but we also provide a default in that case for the sake of
+   * completeness
+   */
+
+  // FIXME P5: this strategy can be improved taking into account full URL. Otherwise, it is
+  // ambigous (e.g. entity which id is "queryContext" and are using a conv op). However, it is
+  // highly improbable that the user uses entities which id (or type) match the name of a
+  // standard operation
+
+  // FIXME P5: unhardwire literals
+  if      (strcasestr(url, "updateContextAvailabilitySubscription") != NULL) return DEFAULT_SERVICE_PATH_RECURSIVE;
+  else if (strcasestr(url, "unsubscribeContextAvailability") != NULL)        return DEFAULT_SERVICE_PATH_RECURSIVE;
+  else if (strcasestr(url, "subscribeContextAvailability") != NULL)          return DEFAULT_SERVICE_PATH_RECURSIVE;
+  else if (strcasestr(url, "discoverContextAvailability") != NULL)           return DEFAULT_SERVICE_PATH_RECURSIVE;
```

Comment:
I'm using case insensitive comparison with strcasestr(), instead of strstr(). Strictly speaking, the URL includes case, but I don't think we could have in the future different operations for e.g. "discoverContextAvailability" and "discovercontextavailability. Thus, I think is ok doing this way.

If you agree, mark this as NTC.